etl/src/main/scala/etl/RecipeParsing.scala

```diff
@@ -79,8 +80,10 @@ object RecipeParsing {
 
   def guessSteps(body: Seq[Element]): Seq[String] = {
     val candidates: Seq[Element] = body.filter(NumberedParagraph.matches)
-    candidates.map(_.text).filter(_.nonEmpty)
+    val stepsFound = candidates.map(_.text).filter(_.nonEmpty)
+    val pattern = new scala.util.matching.Regex("""(^\d+.?\s?)(.*+)""", "number", "step")
```

Comment:
Also, the regular expression only needs to be created once so can be pulled out of the function and placed on the object to ensure this. This is good practice as it can be expensive to compile regular expressions (in Java at least as they are compiled into an internal data structure) so this way we only incur the cost once rather than every time the function is called.